File: app/src/main/java/me/ykrank/s1next/viewmodel/UserViewModel.java
package me.ykrank.s1next.viewmodel;

import android.databinding.BaseObservable;
import android.databinding.Bindable;

import me.ykrank.s1next.BR;
import me.ykrank.s1next.data.User;

public final class UserViewModel extends BaseObservable {

    private final User user = new ObservableUser(this);

    public User getUser() {
        return user;
    }

    @Bindable
    public boolean isSigned() {
        return user.isSigned();
    }

    private static final class ObservableUser extends User {

        private final BaseObservable mBaseObservable;

        public ObservableUser(BaseObservable baseObservable) {
            this.mBaseObservable = baseObservable;
        }

        @Override
        public void setLogged(boolean logged) {
            super.setLogged(logged);
            mBaseObservable.notifyChange();
        }

        @Override
        public void setSigned(boolean b) {
            super.setSigned(b);
            mBaseObservable.notifyPropertyChanged(BR.signed);
        }
    }
}

File: app/src/main/java/me/ykrank/s1next/viewmodel/UserViewModel.kt
package me.ykrank.s1next.viewmodel

import android.databinding.BaseObservable
import android.databinding.Bindable

import me.ykrank.s1next.BR
import me.ykrank.s1next.data.User
import me.ykrank.s1next.data.pref.AppDataPreferencesManager

class UserViewModel(appDataPref: AppDataPreferencesManager) : BaseObservable() {

    val user: User = ObservableUser(this, appDataPref)

    val isSigned: Boolean
        @Bindable
        get() = user.isSigned

    val isAuthor: Boolean
        @Bindable
        get() = user.uid == "223963"

    private class ObservableUser(private val mBaseObservable: BaseObservable,
                                 appDataPref: AppDataPreferencesManager) : User(appDataPref) {

        override var isLogged: Boolean
            get() = super.isLogged
            set(logged) {
                super.isLogged = logged
                mBaseObservable.notifyChange()
            }

        override var isSigned: Boolean
            get() = super.isSigned
            set(b) {
                super.isSigned = b
                mBaseObservable.notifyPropertyChanged(BR.signed)
            }
    }
}
